Source Generator for XamlG and CssG

StephaneDelcroix opened the pull request:
### Description of Change ###

Replace the XamlG build task by a Source Generator.

- this is required for .NET HotReload
- a future Xaml replacement could be based on this


### Additions made ###

No API Change

### PR Checklist ###

<!-- See our [Handler Property PR Guidelines](https://github.com/dotnet/maui/wiki/Handler-Property-PR-Guidelines) for more tips -->

- [x] Targets the correct branch 
- [ ] Tests are passing (or failures are unrelated)
- [ ] Targets a single property for a single control (or intertwined few properties)
- [ ] Adds the property to the appropriate interface
- [ ] Avoids any changes not essential to the handler property
- [ ] Adds the mapping to the PropertyMapper in the handler
- [ ] Adds the mapping method to the Android, iOS, and Standard aspects of the handler
- [ ] Implements the actual property updates (usually in extension methods in the Platform section of Core)
- [ ] Tags ported renderer methods with [PortHandler]
- [ ] Adds an example of the property to the sample project (MainPage)
- [ ] Adds the property to the stub class
- [ ] Implements basic property tests in DeviceTests

#### Does this PR touch anything that might affect accessibility?
- [ ] Does this PR introduce a new control? (If yes, add an example using SemanticProperties to the SemanticsPage)
- [ ] APIs that modify focusability?
- [ ] APIs that modify any text property on a control?
- [ ] Does this PR modify view nesting or view arrangement in anyway?
- [ ] Is there the smallest possibility that your PR will change accessibility? 
- [ ] I'm not sure, please help me

If any of the above checkboxes apply to your PR, then the PR will need to provide testing to demonstrate that accessibility still works.

Redth commented on .nuspec/Microsoft.Maui.Controls.targets:
```diff
@@ -48,50 +47,33 @@
-		</ItemGroup>
-	</Target>
+  <!-- Inject MauiXaml as AdditionalFiles for partial type generation-->  
+  <Target Name="_InjectAdditionalFiles"
```
Let's add more to the target name, maybe a `_MauiXamlInjectAdditionalFiles`

Redth commented on src/Controls/src/SourceGen/CodeBehindGenerator.cs:
```diff
@@ -0,0 +1,333 @@
+				//optional default ctor
+				if (generateDefaultCtor)
+				{
+					sb.AppendLine($"\t\t[global::System.CodeDom.Compiler.GeneratedCodeAttribute(\"Microsoft.Maui.Controls.SourceGen\", \"1.0.0.0\")]");
```
Should we apply the GeneratedCodeAttribute to the class itself too?

StephaneDelcroix replied:
we only add the attribute to the class is it's codebehind for a code-less ResourceDictionary

StephaneDelcroix committed:
[X] Replace XamlG with source Generator

C# HotReload do not support c# files generated through tasks. So we're
moving the XamlG capability to a C# Source Generator

StephaneDelcroix committed:
package the generator

StephaneDelcroix committed:
add the dependencies

StephaneDelcroix committed:
CssG is gone

StephaneDelcroix committed:
exclude wwwroot

StephaneDelcroix commented on .nuspec/Microsoft.Maui.Controls.DefaultItems.props:
```diff
@@ -1,11 +1,11 @@
 
   <ItemGroup Condition="'$(EnableDefaultItems)'=='True' And '$(EnableDefaultCssItems)'=='True' And '$(EnableDefaultEmbeddedResourceItems)'=='True'">
-    <EmbeddedResource Include="**\*.css" Exclude="$(DefaultItemExcludes);$(DefaultExcludesInProjectFolder)" />
+    <MauiCss Include="**\*.css" Exclude="$(DefaultItemExcludes);$(DefaultExcludesInProjectFolder);$(DefaultWebContentItemExcludes)" />
```
DefaultWebContentItemExcludes is to excludes wwwroot

StephaneDelcroix commented on .nuspec/Microsoft.Maui.Controls.targets:
```diff
@@ -1,15 +1,11 @@
 		<EnableDefaultXamlItems Condition="'$(EnableDefaultXamlItems)'==''">True</EnableDefaultXamlItems>
-		<_DefaultXamlItemsEnabled>False</_DefaultXamlItemsEnabled>
 		<EnableDefaultCssItems Condition="'$(EnableDefaultCssItems)'==''">True</EnableDefaultCssItems>
-		<_DefaultCssItemsEnabled>False</_DefaultCssItemsEnabled>
```
those weren't used

StephaneDelcroix commented on src/Controls/tests/Xaml.UnitTests/Controls.Xaml.UnitTests.csproj:
```diff
@@ -1,6 +1,6 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <TargetFramework>net47</TargetFramework>
+    <TargetFramework>net48</TargetFramework>
```
net48 supports source generators

StephaneDelcroix commented on src/Controls/src/SourceGen/Controls.SourceGen-net6.csproj:
```diff
@@ -0,0 +1,38 @@
+  <ItemGroup>
+    <None Include="$(OutputPath)\$(AssemblyName).dll" Pack="true" PackagePath="analyzers/dotnet/cs" Visible="false" />
+    <None Include="$(OutputPath)\Microsoft.Maui.Controls.dll" Pack="true" PackagePath="analyzers/dotnet/cs" Visible="false" />
+    <None Include="$(OutputPath)\Microsoft.Maui.Controls.Xaml.dll" Pack="true" PackagePath="analyzers/dotnet/cs" Visible="false" />
```
all the dependencies of the source generator need to be available in the analyser folder, otherwise the analyser can't start
